### PR TITLE
Correctly map `pkg_index` tokens

### PR DESCRIPTION
**Pull request by iwahbe**
This PR improves the error message when it fails to map tokens (first commit), then removes the `"_"` appended to the end of the prefix so tokens that are of the form `<pkg>_index` can be correctly mapped.

Fixes https://github.com/pulumi/pulumi-terraform-provider/issues/13

### CODE REVIEW

File: dynamic/info.go [hunk 1]
```diff
@@ -84,7 +84,8 @@ func providerInfo(ctx context.Context, p run.Provider, value parameterize.Value)
 		return prov, err
 	}
 
-	err := prov.ComputeTokens(tokens.SingleModule(p.Name()+"_", "index", tokens.MakeStandard(p.Name())))
+	err := prov.ComputeTokens(tokens.SingleModule(
+		prov.GetResourcePrefix(), "index", tokens.MakeStandard(p.Name())))
```
Comment (mjeffryes) [inline dynamic/info.go:88]: I can see how this fixes this specific error, and I can see that specific case tested in the new test, but I'm not sure I follow why we don't need the "_" for other resources?

File: pkg/tfbridge/tokens_test.go [hunk 2]
```diff
@@ -87,6 +87,24 @@ func TestTokensSingleModule(t *testing.T) {
 	}, info.Resources)
 }
 
+func TestTokenWithModuleName(t *testing.T) {
+	info := tfbridge.ProviderInfo{
+		Name: "foo",
+		P: (&schema.Provider{
+			ResourcesMap: schema.ResourceMap{
+				"foo_index": nil,
```
Comment (mjeffryes) [inline pkg/tfbridge/tokens_test.go:95]: should we be testing some other names against this strategy eg. maybe the list fromt he previous test:
```
"foo_fizz_buzz":       {Tok: "foo:index:FizzBuzz"},
"foo_bar_hello_world": {Tok: "foo:index:BarHelloPulumi"},
"foo_bar":             {Tok: "foo:index:Bar"},
```

**Review by mjeffryes — APPROVED** [review]
I trust your familiarity with the naming code to get this correct, but would feel better if there were more testcases checked for the exact strategy used by the dynamic provider.

File: pkg/tfbridge/tokens_test.go (hunk 2 above)
Comment (iwahbe) [reply pkg/tfbridge/tokens_test.go:95]: This strategy is pretty well tested in:

https://github.com/pulumi/pulumi-terraform-bridge/blob/610224f512cdc81b79ef771f36ba6ac9e9d3387c/pkg/tfbridge/tokens_test.go#L36

File: dynamic/info.go (hunk 1 above)
Comment (iwahbe) [reply dynamic/info.go:88]: This is basically an instance of "we were calling `tokens.SingleModule` wrong. It already knows to split on `"_"`.

**Pull request merged**